kani-driver/src/main.rs

```diff
@@ -73,6 +74,7 @@ fn cargokani_main(input_args: Vec<OsString>) -> Result<()> {
         let result = ctx.check_harness(&specialized_obj, &report_dir, harness)?;
         if result == VerificationStatus::Failure {
             failed_harnesses.push(harness);
+            ctx.exe_trace_main(&specialized_obj, harness);
```

Comment:
To me, it seems more intuitive to have the driver `main` call `exe_trace_main` because it's not something that is a part of running CBMC, it's something that comes after. I refactored the code to generate the CBMC output file from the driver `main` and pass it as a parameter to `run_cbmc`.

Reply:
I made the update. It still seems a bit counter-intuitive to call `exe_trace_main` from inside `run_cbmc` (it's not involved in running CBMC), but it does clean up the code, w.r.t the output file.